pkg/generate.go

```diff
@@ -411,6 +492,123 @@ func generateDeploymentPatch(options gitopsv1alpha1.GeneratorOptions, imageName,
 	return &deployment
 }
 
+func generateStatefulSetPatch(options gitopsv1alpha1.GeneratorOptions, imageName, containerName, namespace string) *appsv1.StatefulSet {
+
+	statefulSet := appsv1.StatefulSet{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "StatefulSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      options.Name,
+			Namespace: namespace,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Selector: &v1.LabelSelector{},
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  containerName,
+							Image: imageName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, env := range options.BaseEnvVar {
+		statefulSet.Spec.Template.Spec.Containers[0].Env = append(statefulSet.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  env.Name,
+			Value: env.Value,
+		})
+	}
+
+	// only add the environment env configurations, if a deployment/binding env is not present with the same env name
+	for _, env := range options.OverlayEnvVar {
+		isPresent := false
+		for _, statefulSetEnv := range statefulSet.Spec.Template.Spec.Containers[0].Env {
+			if statefulSetEnv.Name == env.Name {
+				isPresent = true
+				break
+			}
+		}
+
+		if !isPresent {
+			statefulSet.Spec.Template.Spec.Containers[0].Env = append(statefulSet.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+				Name:  env.Name,
+				Value: env.Value,
+			})
+		}
+	}
+
+	if options.Replicas > 0 {
+		replica := int32(options.Replicas)
+		statefulSet.Spec.Replicas = &replica
+	}
+
+	statefulSet.Spec.Template.Spec.Containers[0].Resources = options.Resources
+
+	return &statefulSet
+}
+
+func generateDaemonSetPatch(options gitopsv1alpha1.GeneratorOptions, imageName, containerName, namespace string) *appsv1.DaemonSet {
+
+	statefulSet := appsv1.DaemonSet{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "DaemonSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      options.Name,
+			Namespace: namespace,
+		},
+		Spec: appsv1.DaemonSetSpec{
```

Comment:
no replicas here right? I couldnt find it either

Reply:
That's correct. DaemonSets run on all nodes, so there's no replicas field.